auth0/client.py

```diff
@@ -79,6 +137,79 @@ def get_revoked_users(self, page: Optional[int] = None, per_page: Optional[int]
         revoked_query = 'app_metadata.services.status:"revoked"'
         return self._search_users(revoked_query, page, per_page)
 
+    def get_roles(self,
+                  name_filter: Optional[str] = None,
+                  include_totals: Optional[bool] = None,
+                  page: Optional[int] = None,
+                  per_page: Optional[int] = None) -> list[RoleData] | RolesWithTotals:
+        params = {}
+        if name_filter is not None:
+            params["name_filter"] = name_filter
+        if include_totals is not None:
+            params["include_totals"] = include_totals
+        if page is not None:
+            params["page"] = page
+        if per_page is not None:
+            params["per_page"] = per_page
+        url = f"https://{self.domain}/api/v2/roles"
+        resp = self._client.get(url, params=params)
+        resp.raise_for_status()
+        if include_totals:
+            return RolesWithTotals(**resp.json())
+        else:
+            return self._convert_roles(resp)
+
+    def get_role_by_name(self, name: str) -> RoleData:
+        """
+        Get a role by name.
+        Raises ValueError if not found, or multiple roles are found.
+        """
+        roles: list[RoleData] = self.get_roles(name_filter=name)
+        if len(roles) == 0:
+            raise ValueError(f"Role with name {name} not found.")
+        elif len(roles) > 1:
+            raise ValueError(f"Multiple roles with name {name} found.")
```

Comment:
Would be good to inform how many duplicated roles are found:
```suggestion
            raise ValueError(f" {len(roles)} duplicated roles with name {name} found.")
```

Reply:
👍